src/main/java/com/google/devtools/build/lib/analysis/RunfilesSupport.java

```diff
@@ -441,10 +441,8 @@ public Artifact getRepoMappingManifest() {
   /** Returns the root directory of the runfiles symlink farm; otherwise, returns null. */
   @Nullable
   public Path getRunfilesDirectory() {
-    if (runfilesInputManifest == null) {
-      return null;
-    }
-    return FileSystemUtils.replaceExtension(runfilesInputManifest.getPath(), RUNFILES_DIR_EXT);
+    Path executablePath = owningExecutable.getPath();
+    return executablePath.getParentDirectory().getRelative(executablePath.getBaseName() + RUNFILES_DIR_EXT);
```

Comment:
I haven't tested this, but I think that this should work and be simpler:

```suggestion
    return runfilesTreeArtifact.getPath();
```

Reply:
Great call! I was originally looking off the most recent release branch, which didn't have this. This is cleaner, I'll check it works.